### PR TITLE
Update gPRC for duplicate proto definition

### PR DESCRIPTION
**Pull request by shuo-wu**
https://github.com/longhorn/longhorn/issues/999

### CODE REVIEW

File: generate_grpc.sh [hunk 1]
```diff
@@ -11,15 +11,15 @@ if [ ! -e ./vendor_proto/protobuf/src/google/protobuf ]; then
 fi
 
 # controller
-protoc -I pkg/engine/controller/rpc/pb -I vendor_proto/protobuf/src/ pkg/engine/controller/rpc/pb/controller.proto --go_out=plugins=grpc:pkg/engine/controller/rpc/pb
+protoc -I pkg/engine/controller/rpc/pb -I vendor_proto/protobuf/src/ pkg/engine/controller/rpc/pb/controller.proto --go_out=plugins=grpc:$GOPATH/src/
```
Comment (yasker) [inline generate_grpc.sh]: Can we avoid using GOPATH? There is no guarantee GOPATH exists. With `go mod`, GOPATH can be empty.

Comment (shuo-wu) [reply generate_grpc.sh]: OK. I modified it to use the relative path. And I added comments in the file `syncagent.proto` to explain the reason.

File: generate_grpc.sh [hunk 2]
```diff
@@ -11,15 +11,15 @@ if [ ! -e ./vendor_proto/protobuf/src/google/protobuf ]; then
 fi
 
 # controller
-protoc -I pkg/engine/controller/rpc/pb -I vendor_proto/protobuf/src/ pkg/engine/controller/rpc/pb/controller.proto --go_out=plugins=grpc:pkg/engine/controller/rpc/pb
+protoc -I pkg/engine/controller/rpc/pb -I vendor_proto/protobuf/src/ pkg/engine/controller/rpc/pb/controller.proto --go_out=plugins=grpc:./../../../
```
Comment (yasker) [inline generate_grpc.sh]: This makes less sense than $GOPATH.

If $GOPATH doesn't exist, how can we know what's at `./../../../`?

Comment (shuo-wu) [reply generate_grpc.sh]: As long as the developers follow the convention of go code organization, `./../../../` should be the parent directory of `github.com/`.

Comment (yasker) [reply generate_grpc.sh]: The convention you mentioned is GOPATH. User doesn't need to put directory there using go mod.

**Pull request closed without merging**